src/features/game/gameSlice.ts

```diff
@@ -0,0 +1,477 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { RootState } from '../../app/store';
+import {
+  ExitGameMessage,
+  GotToNextRoundMessage,
+  JoinNewGameMessage,
+  MakeChoiceMessage,
+  PauseGameMessage,
+  ResumeGameMessage,
+  StartNewGameMessage,
+} from './interfaces/messages';
+
+export enum RoundResult {
+  PLAYER_WIN = 'player_win',
+  PLAYER_LOSE = 'player_lose',
+  RESULT_NOT_EXIST = 'result_no_exist',
+  NO_ONE_WIN = 'no_one_win',
+}
+
+export enum Shapes {
+  ROCK = 'rock',
+  SCISSORS = 'scissors',
+  PAPER = 'paper',
+}
+
+type ReduceBestWin = {
+  isPreviousVin: boolean;
+  currentWinCount: number;
+  maxWinCount: number;
+};
+
+export interface PlayerChoice {
+  userSocket: string;
+  choice: Shapes;
+}
+
+export interface GameRound {
+  roundNumber: number;
+  choices: PlayerChoice[];
+  winner: string;
+}
+
+export interface Player {
+  socketId: string;
+  nickName: string;
+  deactivated: string;
+}
+
+export interface Game {
+  _id: string;
+  roundsCount: number;
+  currentRound: number;
+  players: Player[];
+  rounds: GameRound[];
+  paused: boolean;
+  finished: boolean;
+  pausedBy: string;
+}
+
+export interface GameItem {
+  id: string;
+  roundsCount: number;
+}
+
+const NO_ONE_WIN_TEXT = 'Draw!';
```

Comment:
Most probably I would extract messages to a different file, it will help in case of internationalization

Reply:
Moved.